contrib/splitup.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -xe
+
+# VARIABLES
+basedir=$(dirname "$0")
+local_branch=$(cd $basedir && git rev-parse --abbrev-ref HEAD)
+
+
+# FUNCTIONS
+function check_existing_remote {
+  if ! git remote | grep -sq $1; then
+    git remote add $1 git@github.com:/ceph/ansible-ceph-$1.git
+  fi
+}
+
+function pull_origin {
+  git pull origin --tags
+}
+
+function reset_hard_origin {
+  git checkout $local_branch
+  git fetch origin ; git reset --hard origin/master
+}
+
+# MAIN
+pull_origin
+
+for ROLE in $(ls -1 roles | grep -v ceph.ceph-common); do
+  check_existing_remote $ROLE
+  for BRANCH in master ansible-1.9; do
+    if [[ ! -f .git/refs/heads/$BRANCH ]]; then
+      git checkout -t origin/$BRANCH
+    else
+      git checkout $BRANCH
+      git reset --hard origin/$BRANCH
+    fi
```

Comment:
I found L32-36 can be simplified to a single line: `git checkout -B $BRANCH origin/$BRANCH`.

-B will either checkout the remote branch, or it will hard reset the existing local branch to match the remote branch.